### PR TITLE
HDDS-4533.Avoid rewriting pipeline information during PipelineStateManagerV2Impl initialization

### PR DESCRIPTION
**Pull request by amaliujia**
## What changes were proposed in this pull request?

During initialization, PipelineStateManagerV2Impl will rewrites pipeline info into the store. We should avoid it.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4533

## How was this patch tested?

UT

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineStateManagerV2Impl.java [hunk 1]
```diff
@@ -78,7 +78,8 @@ private void initialize() throws IOException {
         iterator = pipelineStore.iterator();
     while (iterator.hasNext()) {
       Pipeline pipeline = iterator.next().getValue();
-      addPipeline(pipeline.getProtobufMessage());
```
Comment (amaliujia) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineStateManagerV2Impl.java:81]: addPipeline will do the following:
```
      Pipeline pipeline = Pipeline.getFromProtobuf(pipelineProto);
      pipelineStore.put(pipeline.getId(), pipeline);
      pipelineStateMap.addPipeline(pipeline);
      nodeManager.addPipeline(pipeline);
```
`pipelineStore.put(pipeline.getId(), pipeline);` should be avoided.

**Comment by amaliujia** [comment]
R: @runzhiwang @GlenGeng @ChenSammi @bshashikant

**Comment by GlenGeng** [comment]
+1 
Waiting for CI. Thanks for quick fix!

**Review by runzhiwang — APPROVED** [review]
LGTM

**Comment by amaliujia** [comment]
@GlenGeng @runzhiwang thanks! CI all green now

**Comment by runzhiwang** [comment]
@amaliujia Thanks the patch. @GlenGeng Thanks for review. I have merged the patch.

**Pull request merged**